packages/components/src/presentation-components/properties/UniquePropertyValuesSelector.tsx

```diff
@@ -127,23 +134,21 @@ function useUniquePropertyValuesRuleset({ descriptor, field, imodel }: UseUnique
         ],
       });
     })();
-  }, [field, descriptor, imodel]);
+  }, [field]);
 
   return ruleset;
 }
 
-async function getSchemaAndClassNames({ imodel, descriptor, field }: UseUniquePropertyValuesRulesetProps) {
+function getBaseClassInfo(field?: Field) {
   if (field?.parent === undefined && field?.isPropertiesField()) {
-    return field.properties[0].property.classInfo.name.split(":");
+    return field.properties[0].property.classInfo;
   }
-  return (
-    await findBaseExpressionClass(
-      imodel,
-      descriptor.selectClasses.map((item) => item.selectClassInfo),
-    )
-  ).name.split(":");
+  const lastStepToPrimaryClass = field?.parent?.pathToPrimaryClass.slice(-1).pop();
```

Comment:
This does not handle the case where `field` is deep nested under multiple fields.
```